setup project structure for frontend and backend

Orest-Haponov opened the pull request.

Orest-Haponov commented:
@nikita-shmatko @omnomnomka 
- Changed backend build from express to nest.js;
- Stabilized dependencies;
- Moved all @types to devDependencies;
- Added scss;
- Updated readme and gitignor

omnomnomka commented on frontend/.eslintcache:
```diff
@@ -0,0 +1 @@
+[{"C:\\sigma-js-internship\\frontend\\src\\index.tsx":"1","C:\\sigma-js-internship\\frontend\\src\\serviceWorker.ts":"2","C:\\sigma-js-internship\\frontend\\src\\app\\App.tsx":"3","C:\\sigma-js-internship\\frontend\\src\\store\\store.ts":"4","C:\\sigma-js-internship\\frontend\\src\\store\\reducer\\rootReducer.ts":"5"},{"size":655,"mtime":1608128454680,"results":"6","hashOfConfig":"7"},{"size":542 ...
```
Please, add this file to gitignore

omnomnomka commented on frontend/public/robots.txt:
```diff
@@ -0,0 +1,3 @@
+# https://www.robotstxt.org/robotstxt.html
```
I don't think this file is necessary for our project

Orest-Haponov commented:
@omnomnomka @nikita-shmatko 
- removed .eslintcache and robots.txt from project
- updated .gitignore (.eslintcache was already in it, but in some reason didn't work)
